tests/functional_test_base.go

```diff
@@ -194,31 +196,32 @@ func (s *FunctionalTestBase) setupLogger() {
 }
 
 // checkTestShard supports test sharding based on environment variables.
-func (s *FunctionalTestBase) checkTestShard() {
+func checkTestShard(t *testing.T) {
 	totalStr := os.Getenv("TEST_TOTAL_SHARDS")
 	indexStr := os.Getenv("TEST_SHARD_INDEX")
 	if totalStr == "" || indexStr == "" {
 		return
 	}
 	total, err := strconv.Atoi(totalStr)
-	s.NoError(err)
-	s.GreaterOrEqual(total, 1)
+	if err != nil || total < 1 {
+		t.Fatal("Couldn't convert TEST_TOTAL_SHARDS")
+	}
 	index, err := strconv.Atoi(indexStr)
-	s.NoError(err)
-	s.GreaterOrEqual(index, 0)
-	s.Less(index, total)
+	if err != nil || index < 0 || index >= total {
+		t.Fatal("Couldn't convert TEST_SHARD_INDEX")
+	}
 
 	// This was determined empirically to distribute our existing test names + run times
 	// reasonably well. This can be adjusted from time to time.
 	// For parallelism 4, use 11. For 3, use 26. For 2, use 20.
 	const salt = "-salt-26"
 
-	nameToHash := s.T().Name() + salt
+	nameToHash := t.Name() + salt
 	testIndex := int(farm.Fingerprint32([]byte(nameToHash))) % total
 	if testIndex != index {
-		s.T().Skipf("Skipping %s in test shard %d/%d (it runs in %d)", s.T().Name(), index, total, testIndex)
+		t.Skipf("Skipping %s in test shard %d/%d (it runs in %d)", t.Name(), index, total, testIndex)
```

Comment:
Wouldn't it be more clear to print `index+1` (1/3, 2/3, 3/3) instead of `index` (0/3, 1/3, 2/3)?

Reply:
Yeah probably :)  I'll do that